code/game/machinery/machinery.dm

```diff
@@ -588,11 +559,16 @@ Class Procs:
 
 	if(iswrench(O) && machine_flags & WRENCHMOVE) //make sure this is BEFORE the fixed2work check
 		if(!panel_open)
-			return wrenchAnchor(user)
+			if(state == 2 && src.machine_flags & WELD_FIXED) //prevent unanchoring welded machinery
+				to_chat(user, "\The [src] has to be unwelded from the floor first.")
+				return -1 //state set to 2, can't do it
+			else
+				if(wrenchAnchor(user) && machine_flags && FIXED2WORK) //wrenches/unwrenches into place if possible, the updates the power
+					power_change() //updates us to turn on or off as necessary
+					return
```

Comment:
You should return 1 after a non-zero wrenchAnchor to show something has happened.